lib/uplink/objects.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package uplink
+
+import (
+	"time"
+
+	"storj.io/storj/pkg/storj"
+)
+
+// ObjectMeta represents metadata about a specific Object
+type ObjectMeta struct {
+	Bucket   string
+	Path     storj.Path
+	IsPrefix bool
+
+	Metadata map[string]string
+
+	Created  time.Time
+	Modified time.Time
+	Expires  time.Time
+
+	Size     int64
+	Checksum string
+
+	// this differs from storj.Object by not having Version (yet), and not
+	// having a Stream embedded. I'm also not sold on splitting ContentType out
+	// from Metadata but shrugemoji.
+}
+
+// ObjectPutOpts controls options about uploading a new Object, if authorized.
+type ObjectPutOpts struct {
```

Comment:
I think there should also be an Object struct with just the `ObjectMeta` on it (I'm just looking at this though, might be wrong https://github.com/storj/storj/pull/847/files)